Sources/Datadog/Tracing/TracingAutoInstrumentation.swift

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal class TracingAutoInstrumentation {
+    static var instance: TracingAutoInstrumentation?
+
+    let tracedHosts: Set<URL>
+    let swizzler: URLSessionSwizzler
+
+    init?(tracedHosts: Set<URL>?) {
+        guard let someHosts = tracedHosts else {
+            return nil
+        }
+        do {
+            self.tracedHosts = someHosts
+            self.swizzler = try URLSessionSwizzler()
+        } catch {
+            userLogger.warn("🔥 Network requests won't be traced automatically for \(String(describing: tracedHosts)): \(error)")
+            developerLogger?.warn("🔥 Network requests won't be traced automatically for \(String(describing: tracedHosts)): \(error)")
+            return nil
```

Comment:
It's missing tests. By convention, we test all user facing features, including printed error messages if possible. See example in:

https://github.com/DataDog/dd-sdk-ios/blob/c93f9687a7e18199d91f6f38e1a9944f1c3ed738/Tests/DatadogTests/Datadog/TracerTests.swift#L571-L588

Reply:
good catch
adding 👍

Reply:
Ah I see 👌. For critical & complex things we may use component injection through protocol and provide throwing mock implementation in tests, but I think this part is not that complex, so it's clear and safe from the code. Let's categorise it as "not easy possible" and keep no coverage for it 👍.

Reply:
💡 we could have tested this by removing `dataTaskWithURL/Request` methods from `URLSession` but `ObjCRuntime` doesn't provide `removeMethod`